src/luxonis_ml/loader/augmentations.py

```diff
@@ -334,4 +389,197 @@ def __init__(
             image_size=image_size,
             augmentations=[a for a in augmentations if a["name"] == "Normalize"],
             keep_aspect_ratio=keep_aspect_ratio,
-        )
\ No newline at end of file
+        )
+
+
+from albumentations.core.transforms_interface import (
+    BoxInternalType,
+    DualTransform,
+    KeypointInternalType,
+)
+from albumentations.core.bbox_utils import denormalize_bbox, normalize_bbox
+
+
+class LetterboxResize(DualTransform):
+    def __init__(
+        self,
+        height: int,
+        width: int,
+        interpolation: int = cv2.INTER_LINEAR,
+        border_value: int = 0,
+        mask_value: int = 0,
+        always_apply: bool = False,
+        p: float = 1.0,
+    ):
+        """Augmentation to apply letterbox resizing to images. Alse transforms
+        masks, bboxes and keypoints to correct shape.
+
+        Args:
+            height (int): Desired height of the output
+            width (int): Desired width of the output
+            interpolation (int, optional): Cv2 flag to specify interpolation used when resizing. Defaults to cv2.INTER_LINEAR.
+            border_value (int, optional): Padding value for images. Defaults to 0.
+            mask_value (int, optional): Padding value for masks. Defaults to 0.
+            always_apply (bool, optional): Defaults to False.
+            p (float, optional): Probability of applying the transform. Defaults to 1.0.
+        """
+        super().__init__(always_apply, p)
+
+        if not (0 <= border_value <= 255):
+            raise ValueError("Border value must be in range [0,255].")
+
+        if not (0 <= mask_value <= 255):
+            raise ValueError("Mask value must be in range [0,255].")
+
+        self.height = height
+        self.width = width
+        self.interpolation = interpolation
+        self.border_value = border_value
+        self.mask_value = mask_value
+
+    def update_params(self, params: Dict[str, Any], **kwargs) -> Dict[str, Any]:
+        params = super().update_params(params, **kwargs)
+
+        img_height = params["rows"]
+        img_width = params["cols"]
+
+        ratio = min(self.height / img_height, self.width / img_width)
+        new_height = int(img_height * ratio)
+        new_width = int(img_width * ratio)
+
+        # only supports center alignment
```

Comment:
Would we want to add support for random position in the future?

Reply:
Albumentations [PadIfNeeded](https://albumentations.ai/docs/api_reference/full_reference/#albumentations.augmentations.geometric.transforms.PadIfNeeded) also supports other positioons (e.g. image in top left corner, pad on right and bottom) so I left the ability to do that in there -  should be just the case of setting the padding values depending on the parameter. Might want to test it out and add support for this as well before merge.